dist_tables/dml.rst

```diff
@@ -38,6 +38,70 @@ For example:
 
     If COPY fails to open a connection for a shard placement then it behaves in the same way as INSERT, namely to mark the placement(s) as inactive unless there are no more active placements. If any other failure occurs after connecting, the transaction is rolled back and thus no metadata changes are made.
 
+INSERT INTO ... SELECT
+$$$$$$$$$$$$$$$$$$$$$$
+
+Citus is commonly used to scale out event data pipelines and populate real-time dashboards. These applications require fast queries across predefined aggregations of incoming data. To achieve sufficient speed with high-volume data, it helps to pre-aggregated data. This is called "rolling up" the data and it avoids the cost of processing raw data at run-time. As an extra benefit rolling up timeseries data into hourly or daily statistics can also save space. When the full details of old data are no longer needed and aggregates suffice then old data may be deleted.
+
+It is important to be able to update rollups *incrementally*, that is, without re-scanning all data. Both the rolling up and updating can be accomplished with :code:`INSERT INTO ... SELECT`, whereby the results of a select query are inserted into a table. A full reference for this feature is available in the PostgreSQL `docs <https://www.postgresql.org/docs/current/static/sql-insert.html>`_. It works well on Citus because it is parallelizable under certain conditions. We'll see more about that later.
+
+First some basics. Insert into select uses the results of a select query to populate a table. For instance consider a simple select statement which counts the number of web page views per page per day.
+
+.. code-block:: postgres
+
+  SELECT view_time::date AS day, page_id, count(*) AS view_count
+  FROM page_views
+  GROUP BY view_time::date, page_id;
+
+We can create a rollup table :code:`daily_page_views` to hold these results and fill it with the results of the previous query:
+
+.. code-block:: postgres
+
+  CREATE TABLE daily_page_views (
+    day date,
+    page_id int,
+    view_count bigint,
+    primary key (day, page_id)
+  );
+
+  INSERT INTO daily_page_views (day, page_id, view_count)
+    -- this is our original select query
+    SELECT view_time::date AS day, page_id, count(*) AS view_count
+    FROM page_views
+    GROUP BY view_time::date, page_id;
+
+We can use PostgreSQL's upsert feature to incrementally update certain parts of the rollup table. For instance suppose we have already rolled up visits happening before a certain timestamp (we'll call it :code:`$1`) and want to update the rollups to include more recent views. Just add a WHERE clause and specify "ON CONFLICT." The latter is PostgreSQL's technique for doing `upserts <https://www.postgresql.org/docs/9.5/static/sql-insert.html#SQL-ON-CONFLICT>`_.
```

Comment:
* My own heuristic is to use "we" when the documentation is guiding the reader through an example, because we (the author and reader) are considering it together. I use "you" less frequently, in places where older English writing would traditionally use "[one](https://en.wikipedia.org/wiki/One_(pronoun))." I could be more consistent in this section though, I'll fix it.
* "Just add a WHERE..." was just sloppy, I'll rewrite it.